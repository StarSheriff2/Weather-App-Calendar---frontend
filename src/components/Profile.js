import React from 'react';
import { Redirect } from 'react-router-dom';
import { useSelector } from 'react-redux';

const Profile = () => {
  const { user: currentUser } = useSelector((state) => state.auth);

  if (!currentUser) {
    return <Redirect to="/" />;
  }

  return (
    <div className="container">
      <header className="jumbotron">
        <h3>
          <strong>{currentUser.name}</strong>
          {' '}
          Profile
        </h3>
      </header>
      <p>
        <strong>Name:</strong>
        {' '}
        {currentUser.name}
      </p>
      <p>
        <strong>Email:</strong>
        {' '}
        {currentUser.email}
      </p>
    </div>
  );
};

export default Profile;
