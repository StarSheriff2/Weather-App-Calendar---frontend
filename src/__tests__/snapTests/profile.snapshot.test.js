import React from 'react';
import { cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { renderWithRedux } from '../../tests/test-utils';
import Profile from '../../components/Profile';

afterEach(cleanup);

describe('Profile', () => {
  test('Renders the Component', () => {
    const profile = renderWithRedux(
      <Router>
        <Profile />
      </Router>,
    ).toJSON();

    expect(profile).toMatchSnapshot();
  });
});
