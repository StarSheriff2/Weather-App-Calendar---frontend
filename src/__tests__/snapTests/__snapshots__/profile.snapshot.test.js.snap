// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Renders the Component 1`] = `null`;
